api/dbadapters/bigquery.ts

```diff
@@ -256,6 +258,61 @@ export class BigQueryDbAdapter implements IDbAdapter {
     await this.runQuery(updateQuery);
   }
 
+  public async setMetadata(action: dataform.IExecutionAction): Promise<void> {
+    const { target, actionDescriptor } = action;
+
+    await this.pool
+      .addSingleTask({
+        generator: async () => {
+          try {
+            const metaData = await this.getMetadata(target);
+            const schemaWithDescription = addDescriptionToMetadata(actionDescriptor?.columns, metaData.schema.fields)
```

Comment:
IMO just return immediately at the start of this function if there is no `actionDescriptor` - it's faster